spec/models/unpublishing_alert_spec.rb

```diff
@@ -0,0 +1,215 @@
+require "spec_helper"
+
+RSpec.describe UnpublishingAlert do
+  include LockHandlerTestHelpers
+  let(:content_id) { SecureRandom.uuid }
+  let(:govuk_request_id) { SecureRandom.uuid }
+  let(:public_updated_at) { updated_now.iso8601 }
+
+  let(:published_in_error_payload) do
+    {
+      "document_type" => "gone",
+      "schema_name" => "gone",
+      "base_path" => "/foo",
+      "locale" => "en",
+      "publishing_app" => "whitehall",
+      "public_updated_at" => public_updated_at,
+      "details" => {
+        "explanation" => "Why the page was published in error",
+        "alternative_path" => alternative_path,
+      },
+      "content_id" => content_id,
+      "govuk_request_id" => govuk_request_id,
+      "payload_version" => 199,
+    }
+  end
+
+  let(:consolidated_payload) do
+    {
+      "document_type" => "redirect",
+      "schema_name" => "redirect",
+      "base_path" => "/foo",
+      "locale" => "en",
+      "publishing_app" => "whitehall",
+      "redirects": [
+        {
+          "path" => "/government/publications/foo/html-attachment-foo",
+          "type" => "exact",
+          "destination" => "/government/publications/foobar",
+        },
+      ],
+      "public_updated_at" => public_updated_at,
+      "content_id" => content_id,
+      "govuk_request_id" => govuk_request_id,
+      "payload_version" => 199,
+    }
+  end
+
+  let(:fake_lock_handler_class) do
+    Class.new do
+      def with_lock_unless_done
+        yield
+      end
+    end
+  end
+
+  let(:logger) { double(:logger, info: nil) }
+  let(:unpublishing_alert) { UnpublishingAlert.new(document, logger, unpublishing_scenario) }
+  let(:fake_lock_handler) { fake_lock_handler_class.new }
+  # replace with webmock after GDS API adapters merged
+  let(:alert_api) { double(:alert_api, bulk_unsubscribe: nil) }
+
+  before :each do
+    allow(LockHandler).to receive(:new).and_return(fake_lock_handler)
+    # replace with webmock after GDS API adapters merged?
```

Comment:
I agree we should do this - it'll be much easier for future people working on the app if it uses the same method for setting tests up as most other things.

Reply:
Yeah Michael and I had a chat about it - ended up on the old "Should we stick to the norm for the repo, or what we think is best pratices for the platform".

We ended up saying it might be more confusing to have one test file working different from the others, which sort of calls for a broader refactor 🤔